feat(Evaluation Metrics Node): Add Evaluation Metrics node (no-changelog)

OlegIvaniv opened the pull request:
## Summary

This PR implements new `Evaluation Metrics` used to send metrics data for evaluation workflows. It also adds `disableType` and `defaultType` to `AssignmentTypeOptions` effectively working as Set node with fixed `number` type.

## Related Linear tickets, Github issues, and Community forum posts

<!--
Include links to **Linear ticket** or Github issue or Community forum post.
Important in order to close *automatically* and provide context to reviewers.
-->
<!-- Use "closes #<issue-number>", "fixes #<issue-number>", or "resolves #<issue-number>" to automatically close issues when the PR is merged. -->


## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

OlegIvaniv committed:
feat: Add Evaluation Metrics node and support for default assignment type

OlegIvaniv committed:
Add basic tests

OlegIvaniv committed:
Make metrics plural

jeanpaul reviewed (changes requested):
Looks good, but I can still set some stuff to string if I use drag & drop (see my comment in-line); and then you can't change it anymore. Should we just really force this field to be number (or always cast the value to number?)

jeanpaul commented:
Oh, and it's a bit odd to have it be called `output.politeness` -- I only figured this out when I ran the test. Could we just use the final part of the field for the name of the metric?

https://github.com/user-attachments/assets/d768654e-931b-49f1-8236-a0a1ebd61e1a

OlegIvaniv committed:
Merge branch 'master' into ai-769-add-metrics-node_rebased

OlegIvaniv committed:
Add string to number conversion and error handling for invalid values

OlegIvaniv committed:
Enhance defaultType handling in assignments

- Updated defaultType prop to be undefined by default.
- Added tests to verify defaultType behavior for new assignments, including drag-and-drop scenarios.
- Ensured that defaultType is respected when disableType is true and for all assignments when specified.

OlegIvaniv commented:
> Oh, and it's a bit odd to have it be called `output.politeness` -- I only figured this out when I ran the test. Could we just use the final part of the field for the name of the metric?
> 
>  Screen.Recording.2025-03-20.at.15.34.49.mov

I didn't include fix for this one as it's using the same logic as for the Set node so not easy to override. We could add it as typeOption to `assignments` parameter type but I'd suggest to do that in another PR

jeanpaul reviewed (approved):
🚀

janober commented:
Got released with [`n8n@1.86.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.86.0)
